sdk/python/smithy/dbmanagers/db_context.py

```diff
@@ -0,0 +1,125 @@
+from contextlib import AbstractContextManager
+from typing import List, Optional, Iterator
+from logging import Logger
+
+from smithy.dbmanagers.db_manager import DBManager
+from smithy.remote_store.findings_service.v1 import findings_service_pb2 as pb2
+
+
+class DBContext(AbstractContextManager, Iterator):
+    """
+    DBContext is a context manager for database operations.
+    It provides methods to get findings from the database with pagination support.
+    """
+
+    def __init__(self, db_manager: DBManager, logger: Logger) -> None:
+        """
+        Initialize the DBContext.
+
+        :param db_manager: The database manager instance to be used for database operations.
+        :type db_manager: DBManager
+
+        :param logger: An instance of Logger for logging messages.
+        :type logger: Logger
+
+        :raises TypeError: If the db_manager is not an instance of DBManager.
+        """
+
+        if not isinstance(db_manager, DBManager):
+            raise TypeError("db_manager must be an instance of DBManager.")
+
+        self.db_manager = db_manager
+        self._iter: Optional[Iterator] = None
+        self._pending_updates: List = []
+        self._page = 0
+        self._has_more: bool = True
+        self._log = logger
+
+    def __enter__(self) -> "DBContext":
+        """
+        Enter the context manager, returning the instance itself.
+
+        :return: The instance of DBContext.
+        :rtype: DBContext
+        """
+        self._page = 0
+        self._load_next_page()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> bool:
+        """
+        Flush any queued updates, then close resources.
+        Exceptions propagate (`return False`) unless you decide otherwise.
+        """
+
+        if self._pending_updates:
+            self._flush_updates()
+
+        if hasattr(self.db_manager, "close"):
```

Comment:
this can be simplified by forcing DBManager subclasses to have the `close` method and they just implement a noop if they don't need to have something